ada_2_fhir-r4/zibs2020/payload/0.0/nl-core-HealthcareProvider.xsl

```diff
@@ -0,0 +1,188 @@
+            <xsl:variable name="uuidNode" as="element()" select="."/>
+
+            <Organization>
+                <id value="{nf:get-uuid($uuidNode)}"/>
+                <meta>
+                    <profile value="http://nictiz.nl/fhir/StructureDefinition/nl-core-HealthcareProvider-Organization"/>
+                </meta>
+                <xsl:for-each select="zorgaanbieder_identificatienummer[@value]">
+                    <identifier>
+                        <xsl:call-template name="id-to-Identifier">
+                            <xsl:with-param name="in" select="."/>
+                        </xsl:call-template>
+                    </identifier>
+                </xsl:for-each>
+                <!-- type -->
+                <xsl:for-each select="organisatie_type | afdeling_specialisme">
+                    <type>
+                        <xsl:call-template name="code-to-CodeableConcept">
+                            <xsl:with-param name="in" select="."/>
+                        </xsl:call-template>
+                    </type>
+                </xsl:for-each>
+                <!-- name -->
+                <xsl:if test="organisatie_naam/@value">
+                    <name value="{organisatie_naam/@value}"/>
+                </xsl:if>
+                <!-- contactgegevens -->
+                <!--<xsl:call-template name="nl-core-ContactInformation">
+                    <xsl:with-param name="in" select="contactgegevens"/>
+                </xsl:call-template>-->
+                <!-- address -->
+                <!--<xsl:call-template name="nl-core-AddressInformation">
+                    <xsl:with-param name="in" select="adresgegevens"/>
+                </xsl:call-template>-->
+            </Organization>
+        </xsl:for-each>
+    </xsl:template>
+    
+    <xd:doc>
+        <xd:desc>Creates nl-core-HealthcareProvider-Organization reference</xd:desc>
+    </xd:doc>
+    <xsl:template name="nl-core-HealthcareProvider-Organization-reference" match="zorgaanbieder[not(zorgaanbieder)]" mode="nl-core-HealthcareProvider-Organization-reference" as="element(f:reference)">
+        <xsl:param name="in" select="." as="element()?"/>
+        <xsl:param name="logicalId" as="xs:string?"/>
+        <xsl:param name="fullUrl" as="xs:string?"/>
+        
+        <!-- Because we need to generate a unique uuid, we slightly edit the ada input to make the input node unique -->
+        <xsl:variable name="uuidNode" as="element()" select="."/>
+        
+        <xsl:variable name="identifier" select="$in/zorgaanbieder_identificatienummer[normalize-space(@value | @nullFlavor)]"/>
+        <xsl:variable name="logicalIdOrUuid">
+            <xsl:choose>
+                <xsl:when test="$logicalId">
+                    <xsl:value-of select="$logicalId"/>
+                </xsl:when>
+                <xsl:otherwise>
+                    <xsl:value-of select="nf:get-uuid($in)"/>
```

Comment:
3/3

Hoe verzinnen we iets wat:

1. Gegarandeerd een unieke id geeft als referById true() is
2. Maar ook herhaaldelijk dezelfde id teruggeeft als een template meerdere keren wordt aangeroepen
3. Maar ook vanuit 1 ada-element meerdere verschillende resources met unieke uuids kan genereren.

Ik denk dat de STU3-oplossing -> 'vantevoren' de ada-elementen verrijken hier eigenlijk de enige oplossing is.